src/Appwrite/Platform/Tasks/schedule.php

```diff
@@ -1,231 +1,221 @@
+                    $delayedExecutions = []; // Group executions with same delay to share one coroutine
+
+                    foreach ($schedules as $key => $schedule) {
+                        $cron = new CronExpression($schedule['schedule']);
+                        $nextDate = $cron->getNextRunDate();
+                        $next = DateTime::format($nextDate);
+
+                        $currentTick = $next < $timeFrame;
+
+                        if (!$currentTick) {
                             continue;
                         }
 
-                        $cron = new CronExpression($function['schedule']);
-                        $next = DateTime::format($cron->getNextRunDate());
-
-                        /**
-                        * If next schedule is in 5-min timeframe
-                        * and it was not removed or changed, re-enqueue the function.
-                        */
-                        if (
-                            $next < $timeFrame &&
-                            $function['schedule'] ?? [] === $functions[$function['resourceId']]['schedule']
-                        ) {
-                            $queue[$next][$function['resourceId']] = $function;
+                        $total++;
+
+                        $promiseStart = \microtime(true); // in seconds
+                        $executionStart = $nextDate->getTimestamp(); // in seconds
+                        $executionSleep = $executionStart - $promiseStart; // Time to wait from now until execution needs to be queued
+
+                        $delay = \ceil(\intval($executionSleep));
+
+                        if (!isset($delayedExecutions[$delay])) {
+                            $delayedExecutions[$delay] = [];
                         }
-                        unset($queue[$slot][$function['resourceId']]); /** removing function from slot */
-                        $count++;
+
+                        $delayedExecutions[$delay][] = $key;
                     }
-                    unset($queue[$slot]); /** removing slot */
-                }
 
-                $timerEnd = \microtime(true);
-                Console::info("Queue timer: finished in " . ($timerEnd - $timerStart) . " seconds with {$count} functions");
-            });
-        }
-    );
-});
+                    foreach ($delayedExecutions as $delay => $scheduleKeys) {
+                        \go(function () use ($delay, $schedules, $scheduleKeys) {
+                            \sleep($delay); // in seconds
+
+                            foreach ($scheduleKeys as $scheduleKey) {
+                                // Ensure schedule was not deleted
+                                if (!isset($schedules[$scheduleKey])) {
+                                    return;
+                                }
+
+                                Console::success("Executing function at " . DateTime::now()); // TODO: Send to worker queue
```

Comment:
We need to add the logic here to send the execution to the worker

Reply:
That is part of this PR: https://github.com/appwrite/appwrite/pull/4659/files#diff-296e1ba7bae39c587087aeeb5dfbfec006980c935a599fb0903c2c2bb1063541R184-R191

Reply:
Yes but when you merge it to that branch it will be removed since your PR does not contain that changes